package.json

```diff
@@ -15,7 +15,7 @@
   ],
   "scripts": {
     "clean": "rimraf lib dist es coverage",
-    "lint": "eslint src test build",
+    "lint": "eslint src test build && prettier --write '{src,test}/**/*.js'",
```

Comment:
I might extract this to a separate script. The act of cleaning up code and linting it for errors is really two separate things. In fact, it probably should be in reverse order so that the linting is on the transformed code. I'd extract it out and add it to the `prepare` script before the lint command.

Reply:
This doesn't error out if there were changes though. Isn't it better to use `--list-different` which errors out? The [prettier documentation](https://github.com/prettier/prettier#--list-different) specifically mentions this flag for CI purposes.

Reply:
What if we used both `--write` and `--list-different`? As you said, `--list-different` will exit non-zero when `prettier` finds that the formatting of at least one file was different than what it should be. And `--write` will actually fix that formatting so users don't have to run it themselves (one less iteration for a redux dev). All this while still exiting non-zero so CI will fail if there were any formatting changes.

Reply:
Wouldn't this be confusing if you run the format command and get errors thrown when you actually expect them to be fixed in place? 

In the spirit of redux I'd prefer the approach of @timdorr  to create an extra command (or rather two). One for formatting in place for development and one side-effect free that just checks for errors for CI. This would also make more sense since we don't run eslint with the `--fix` flag

Reply:
@eps1lon, it doesn't actually throw errors per se, it just exits non-zero printing what files were different/changed:
```bash
➜  redux git:(add_prettier) ./node_modules/.bin/prettier --write --list-different '{src,test}/**/*.js'
src/applyMiddleware.js
➜  redux git:(add_prettier) echo $?
1
```
@eps1lon, so you're suggesting a side-effect free addition to the lint command (something like `prettier --list-different '{src,test}/**/*.js'`), *and* a `format` helper utility run script, that devs can run during dev time? 

@timdorr, you're suggesting no addition to the `lint` script, but creating the same `format` script and having it run before `lint` in the `prepare` script?

Those seem a little contradictory, but I'm probably misunderstanding something 😅

Reply:
One `format` script entry with `--write` for devs and one `format:check` with `--list-different`. And 
 add `format:check` to the CI scripts.